homeassistant/components/zwave/climate.py

```diff
@@ -154,10 +161,27 @@ def __init__(self, values, temp_unit):
                     self._zxt_120 = 1
         self.update_properties()
 
+    def _current_mode_setpoints(self):
+        if self.hvac_mode == HVAC_MODE_OFF:
+            return ()
+        if self.hvac_mode == HVAC_MODE_HEAT:
+            return (self.values.setpoint_heating,)
+        if self.hvac_mode == HVAC_MODE_COOL:
+            return (self.values.setpoint_cooling,)
+        if self.hvac_mode == HVAC_MODE_HEAT_COOL:
+            return (self.values.setpoint_heating, self.values.setpoint_cooling)
+        return ()
+
     @property
     def supported_features(self):
         """Return the list of supported features."""
-        support = SUPPORT_TARGET_TEMPERATURE
+        support = 0
+        current_setpoints = self._current_mode_setpoints()
+        if len(current_setpoints) == 1:
```

Comment:
Can `current_setpoints` change during entity lifetime? Supported features are not allowed to change during entity lifetime.

Reply:
Yep. It depends on the current mode currently.
Let me check how UI/alexa/google components interact with climate component.
My main concern is exposing the correct number of setpoints based on the current mode.

Reply:
The target temperature properties should return `None` if the current mode isn't using a corresponding target temperature. See smartthings climate platform eg:
https://github.com/home-assistant/home-assistant/blob/7d2a8b81370fdf643ce059e8f53d831de442eeaa/homeassistant/components/smartthings/climate.py#L297-L323

Reply:
Makes sense. I'll update a PR. Thanks!

Reply:
There are also thermostats (like Eurotronic Spirit-Z) which have two heating setpoints. One for normal and one for Eco mode. So this code should maybe not be limited to cooling?

Reply:
I've updated the PR to address the comments